lightning/src/onion_message/functional_tests.rs

```diff
@@ -138,16 +159,23 @@ impl Drop for TestCustomMessageHandler {
 impl CustomOnionMessageHandler for TestCustomMessageHandler {
 	type CustomMessage = TestCustomMessage;
 	fn handle_custom_message(&self, msg: Self::CustomMessage, responder: Option<Responder>) -> ResponseInstruction<Self::CustomMessage> {
-		match self.expected_messages.lock().unwrap().pop_front() {
-			Some(expected_msg) => assert_eq!(expected_msg, msg),
+		let add_reply_path = match self.expected_messages.lock().unwrap().pop_front() {
+			Some(Expectations { message: expected_msg, add_reply_path }) => {
+				assert_eq!(expected_msg, msg);
+				add_reply_path
+			},
 			None => panic!("Unexpected message: {:?}", msg),
-		}
+		};
 		let response_option = match msg {
 			TestCustomMessage::Ping => Some(TestCustomMessage::Pong),
-			TestCustomMessage::Pong => None,
+			TestCustomMessage::Pong => Some(TestCustomMessage::Ping),
 		};
 		if let (Some(response), Some(responder)) = (response_option, responder) {
-			responder.respond(response)
+			if add_reply_path {
+				responder.respond_with_reply_path(response)
+			} else {
+				responder.respond(response)
+			}
 		} else {
 			ResponseInstruction::NoResponse
 		}
```

Comment:
If the response can no longer be `None`, then you don't need to use an `Option`.

Reply:
Makes sense!
Refactored the code in [e6effd4](https://github.com/lightningdevkit/rust-lightning/commit/e6effd447d47e1c3c7771935bea58801199b60b3)
Thanks!